xy/convex_hull_test.go

```diff
@@ -174,6 +174,34 @@ func TestOctRing(t *testing.T) {
 	}
 }
 
+func TestOctRingPanic(t *testing.T) {
+	flat := []float64{
+		1, 1,
+		1, 2,
+		2, 3,
+		3, 3,
+		4, 2,
+		4, 1,
+		3, 0,
+		2, 0,
+	}
+	calc := convexHullCalculator{layout: geom.XYM, stride: 2}
+	got := calc.computeOctRing(flat)
+	expected := []float64{
+		1, 1,
+		1, 2,
+		2, 3,
+		3, 3,
+		4, 2,
+		4, 1,
+		3, 0,
+		1, 1,
+	}
+	if !reflect.DeepEqual(expected, got) {
+		t.Fatalf("calc.computeOctRing failed. Expected \n\t%v\nbut was \n\t%v", expected, got)
+	}
```

Comment:
This test uses non-exported functions, and tests that `calc.computeOctRing(flat)` replaces the last coordinate with the first coordinate, throwing away the last coordinate in the process. This is almost certainly not correct.